code/datums/components/_component.dm

```diff
@@ -123,20 +123,24 @@
  * Internal proc to handle behaviour when being removed from a parent
  */
 /datum/component/proc/_RemoveFromParent()
-	var/datum/P = parent
-	var/list/dc = P.datum_components
+	var/datum/parent = src.parent
```

Comment:
why explicit src here?

Reply:
because otherwise you cant name the local variable the same as the datum variable, and i hate reading code where its harder to read just because a datum variable im familiar with was renamed in a local var for performance purposes

Reply:
sure